### PR TITLE
ImportC: add Digital Mars C bitfield support

### PR DESCRIPTION
**Pull request by WalterBright**
Account for variation in DMC bit field implementation from MSC.

This should complete ImportC's bit field support. Yay!

### CODE REVIEW

**Comment by WalterBright** [comment]
Hmm, the failure is when -mscoff32 switch is thrown (which then turns of Microsoft C behavior). How do I stop that from happening?

**Comment by WalterBright** [comment]
Azure is bound and determined to add Microsoft switches like `-mscrtlib=msvcrt120 "-L/LIBPATH:D:\a\1\s\mingw\dmd2\windows\lib32mscoff\mingw" (-O -inline -g)` which of course cause the DigitalMars test to fail. Sigh.

**Comment by WalterBright** [comment]
Had to disable the test because of Azure. See https://issues.dlang.org/show_bug.cgi?id=22257

File: test/runnable/bitfieldsdm.c [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+A2  = 12 4 || 12 4 | 12 4 | 12 4 | 12 4
+A3  = 16 4 || 16 4 | 16 4 | 16 4 | 16 4
+A4  = 12 4 || 12 4 | 12 4 |  8 4 |  8 4
+A5  =  2 1 ||  2 1 |  2 1 |  2 1 |  2 1
+A6  =  4 2 ||  4 2 |  4 2 |  2 2 |  2 2
+A7  = 16 4 || 16 4 | 16 4 | 12 4 | 16 8
+A8  = 12 4 || 12 4 | 12 4 |  8 4 |  8 8
+A9  = 32 8 || 32 8 | 32 8 | 16 4 | 16 8
+A10 =  4 2 ||  4 2 |  4 2 |  2 2 |  2 2
+A11 = 16 4 || 15 4 | 16 4 | 12 4 | 12 4
```
[the remark this answers is not in the document]
Comment (WalterBright) [reply test/runnable/bitfieldsdm.c]: typo. It's supposed to be 16 4

**Review by ibuclaw — APPROVED** [review]

**Comment by ibuclaw** [comment]
@WalterBright as 2.098-beta has already branched, I'll cherry-pick the last importC fixes into stable.  Can you make all future importC PRs target stable until 2.098 is released proper?

**Comment by WalterBright** [comment]
ok

**Comment by rainers** [comment]
> Hmm, the failure is when -mscoff32 switch is thrown (which then turns on Microsoft C behavior). How do I stop that from happening?

How about adding `version (CRuntime_DigitalMars)` to the test?

**Comment by WalterBright** [comment]
> How about adding version (CRuntime_DigitalMars) to the test?

versions have no effect on C code

**Comment by WalterBright** [comment]
@ibuclaw any other barriers to ImportC being usable?

**Comment by ibuclaw** [comment]
> @ibuclaw any other barriers to ImportC being usable?

For getting us off the ground, we're probably fine.  C asm support could be a little less D-like, i.e:
```
/* casm.c */
void test()
{
    /* CParser accepts D attributes in asm. */
    asm @safe nothrow { nop; }
}
```
I need to sort out the GCC-iasm parser to reject D-style too, though I can delay that for a couple months.

Otherwise, feel free to have a look at any of the open bug reports though https://issues.dlang.org/buglist.cgi?quicksearch=importC&list_id=237304

**Pull request merged**